astropy/io/fits/_utils.pyx

```diff
@@ -0,0 +1,54 @@
+from collections import OrderedDict
+
+cdef Py_ssize_t BLOCK_SIZE = 2880  # the FITS block size
+cdef Py_ssize_t CARD_LENGTH = 80
+cdef str VALUE_INDICATOR = '= '  # The standard FITS value indicator
+cdef str END_CARD = 'END' + ' ' * 77
+
+
+def parse_header(fileobj):
+    """The main method allowing to parse quickly a FITS header."""
+
+    cards = OrderedDict()
+    cdef read_blocks = []
```

Comment:
I actually though `cdef list` - it's amazing that it worked 😮

Reply:
Oups! It wasn't done on purpose but yeah it seems that cython understand this. The generated code is slightly different: 
```
+14:     cdef read_blocks = []

  __pyx_t_1 = PyList_New(0); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 14, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __pyx_v_read_blocks = __pyx_t_1;
  __pyx_t_1 = 0;
...
+29:         read_blocks.append(block_str)

    __pyx_t_9 = __Pyx_PyObject_Append(__pyx_v_read_blocks, __pyx_v_block_str); if (unlikely(__pyx_t_9 == ((int)-1))) __PYX_ERR(0, 29, __pyx_L1_error)
```
and with `cdef list`:
```
+14:     cdef list read_blocks = []

  __pyx_t_1 = PyList_New(0); if (unlikely(!__pyx_t_1)) __PYX_ERR(0, 14, __pyx_L1_error)
  __Pyx_GOTREF(__pyx_t_1);
  __pyx_v_read_blocks = ((PyObject*)__pyx_t_1);
  __pyx_t_1 = 0;
...
+29:         read_blocks.append(block_str)

    __pyx_t_9 = __Pyx_PyList_Append(__pyx_v_read_blocks, __pyx_v_block_str); if (unlikely(__pyx_t_9 == ((int)-1))) __PYX_ERR(0, 29, __pyx_L1_error)
```